contracts/bonding/BondingManager.sol

```diff
@@ -351,141 +303,220 @@ contract BondingManager is IBondingManager, Controllable {
-
-        // Mint token reward and allocate to this protocol contract
+        /* // Mint token reward and allocate to this protocol contract */
         token.mint(this, mintedTokens);
 
         // Compute transcoder share of minted tokens
-        uint256 transcoderRewardShare = mintedTokens.mul(transcoders[msg.sender].blockRewardCut).div(100);
-
-        // Add remaining rewards (after transcoder share) for the current cycle of the current round to reward multiplier numerator
-        uint256[2] rewardMultiplier = rewardMultiplierPerTranscoderAndRound[msg.sender][currentRound];
-        rewardMultiplier[0] = rewardMultiplier[0].add(mintedTokens.sub(transcoderRewardShare));
+        uint256 transcoderRewardShare = mintedTokens.mul(t.rates[0]).div(100);
+        // Update transcoder's reward pool for the current round
+        RewardPool storage rewardPool = t.tokenPoolsPerRound[currentRound].rewardPool;
+        rewardPool.rewards = rewardPool.rewards.add(mintedTokens.sub(transcoderRewardShare));
 
-        if (cycleNum == 0 || rewardMultiplier[1] == 0) {
-            // First cycle of current round or reward multiplier denominator has not been set yet
-            // Set transcoder cumulative stake for current round as denominator of reward multiplier for current round
-            rewardMultiplier[1] = activeTranscoderTotalStake(msg.sender);
+        if (rewardPool.transcoderTotalStake == 0) {
+            rewardPool.transcoderTotalStake = transcoderTotalStake(msg.sender);
         }
 
-        rewardMultiplierPerTranscoderAndRound[msg.sender][currentRound] = rewardMultiplier;
-
         // Update transcoder stake with share of minted tokens
-        transcoders[msg.sender].bondedAmount = transcoders[msg.sender].bondedAmount.add(transcoderRewardShare);
-
+        t.bondedAmount = t.bondedAmount.add(transcoderRewardShare);
         // Update transcoder total bonded stake with minted tokens
         transcoderPools.increaseTranscoderStake(msg.sender, mintedTokens);
 
         return true;
     }
 
     /*
-     * @dev Return number of minted tokens for a reward call
+     * @dev Update transcoder's fee pool
+     * @param _transcoder Transcoder address
+     * @param _fees Fees from verified job claims
      */
-    function mintedTokensPerReward() public constant returns (uint256) {
-        return initialTokenSupply.mul(initialYearlyInflation).div(100).div(roundsManager().rewardCallsPerYear());
+    function updateTranscoderFeePool(
+        address _transcoder,
+        uint256 _fees,
+        uint256 _claimBlock,
+        uint256 _transcoderTotalStake
+    )
+        external
+        onlyJobsManager
+        returns (bool)
+    {
+        Transcoder storage t = transcoders[_transcoder];
+
+        // Transcoder must be valid
+        require(t.initialized);
+
+        uint256 currentRound = roundsManager().currentRound();
+        uint256 delegatorsFeeShare = _fees.mul(t.rates[1]).div(100);
```

Comment:
I think we can use a struct here and reference each rate by field name i.e. `rates.blockRewardCut`, `rates.feeShare`, etc.